### PR TITLE
Remove deprecated code

### PR DESCRIPTION
**Pull request by preetapan**
Removed upgrade path code and tests related to: 
- Namespaces (Available since 0.7.0)
- SubmitTime/Version (0.6.0)
- Job Summaries (0.4.0)

### CODE REVIEW

File: command/job_status.go [hunk 1]
```diff
@@ -169,7 +169,7 @@ func (c *JobStatusCommand) Run(args []string) int {
 	basic := []string{
 		fmt.Sprintf("ID|%s", *job.ID),
 		fmt.Sprintf("Name|%s", *job.Name),
-		fmt.Sprintf("Submit Date|%s", formatTime(getSubmitTime(job))),
+		fmt.Sprintf("Submit Date|%s", formatTime(time.Unix(0, *job.SubmitTime))),
```
Comment (notnoop) [inline command/job_status.go]: I like the utility function here, as it's not obvious that `job.Submit` is in nano seconds.

**Comment by preetapan** [comment]
> overall look good to me but some questions to understand implication:
> 
> * Should "Remove in 0.9" sections be removed right before 0.10 or anytime after 0.9.0?  Given that the drain request format changed in 0.8, it'd feel a bit odd dropping support for old format in 0.9.4.
That's fair, perhaps one major version is too short of a deprecation cycle for the drain. I could roll back these changes. In 0.9.4 we could start emitting a warning when servicing requests that use the old format. .
> * For the FSM 0.7 compatibility changes - does dropping these means long running nomad servers must at least snapshot once between upgrading from 0.7 to 0.9.4?
Not necessarily, as long as they hit any of the raft transactions that fixed up missing namespaces in old objects.

**Comment by notnoop** [comment]
> Nomad 0.7 added namespaces and introduced code to fix up older state store objects that were missing namespaces

Cool, let me clarify: Would nomad 0.7 edit the raft log old persisted entries in place in file and how do we do that?  I was expecting Nomad 0.7 to parse old log entries and populate missing namespace fields in memory, and then storing updated entries to raft snapshot without updating old entries.

**Comment by preetapan** [comment]
>Would nomad 0.7 edit the raft log old persisted entries in place in file and how do we do that? I was expecting Nomad 0.7 to parse old log entries and populate missing namespace fields in memory, and then storing updated entries to raft snapshot without updating old entries.

AFAICT we use the [canonicalize](https://github.com/hashicorp/nomad/blob/913427428a43be42b4f23f7d7f1f27ab3f406665/nomad/fsm.go#L1126) method in restore to fix up jobs, but we dont do that consistently for all objects. The code I removed in various upsert methods would handle processing old log entries - when upserting those entries into the state store Nomad 0.7 would add the namespace field. Once we are past the snapshot threshold (8192 entries) we would take a snapshot which saves the statestore (now with namespace in it) to disk, and delete old log entries.

**Pull request merged**